### DISCUSSION TITLE
How to show log messages

### DISCUSSION THREAD

**Discussion by dcoredump** [probonopd/MiniDexed#192] [Q&A] [answered]
Hi, 

I have seen that there is the posibility to add log messages to code `(LOGNOTE()`, `LOGDBG()`, ...). I wonder how to enable showing those messages (e.g. how to set the log level for showing parts or all of those messages)?

Another question: it seems that all `LOG*()` functions add automaticly a newline. Is there a way to supress this behavior?

Thanks, Holger

**Comment by probonopd** [comment]
@rsta2 ^^

**Comment by rsta2** [comment, marked as answer]
There is an option [loglevel=](https://github.com/rsta2/circle/blob/master/doc/cmdline.txt#L13) in the file *cmdline.txt* on the SD card, which controls the amount of generated log messages. The newline cannot be suppressed. Please note that in MiniDexed, log messages will not be displayed from `IRQ_LEVEL` by default, because it uses the `REALTIME` option. For debugging purpose you may have to disable this option in *build.sh* before build, if you want to see messages from IRQ handlers.

**Reply by probonopd** [reply]
Adding to the Developer section of the wiki. Thanks!
